packages/image_picker/example/android/app/src/test/java/io/flutter/plugins/imagepicker/ImageResizerTest.java

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.imagepicker;
+
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+import android.graphics.Bitmap;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.mockito.invocation.InvocationOnMock;
+
+public class ImageResizerTest {
```

Comment:
If we have some time to improve this, it would be better to change the style of testing here so that it's really testing the behavior of `ImageResizer`'s public API instead of the interactions of a bunch of its private methods. The way this test is written right now it's really just verifying that when certain methods are called, other methods also get triggered. This isn't really testing the logic of the class, and it could be broken by trivial method renames. It also exposes a bunch of methods with `@VisibleForTesting` that would be better kept private.

https://testing.googleblog.com/2013/08/testing-on-toilet-test-behavior-not.html

Here's the approach I would probably take:

1. Use [TemporaryFolder](https://junit.org/junit4/javadoc/latest/org/junit/rules/TemporaryFolder.html) to create a file directory for the unit tests, and pass it into `ImageResizer` as `externalFilesDirectory` as its constructed in the tests.
2. Write a sample bitmap to `TemporaryFolder`.
3. Call `ImageResizer.resizeImageIfNeeded` with a combination of settings. Verify that the bitmap at the returned path exists and has been resized correctly for each one.

What do you think, does that sound workable to you? The catch here is I'm not sure how creating a Bitmap in a test goes, and I'm not sure if it's easy to check a Bitmap's image quality or not. I don't remember that API off the top of my head.

This is still an improvement over what we have so I am happy to merge this if needed, thank you for writing tests.

Reply:
Will take a look